model_service/models/opt/model.py

```diff
@@ -163,22 +145,72 @@ def rank(self):
 
     @batch
     def infer(self, **inputs):
-        """Generate sequences from a prompt"""
+        """ Dispatch request to a handler function based on the task """
         self.load_default_args("generate")
-        response = self.generate(inputs)
-        logger.info(f"Infer generation response: {response}")
+
+        task = inputs['task'][0][0].decode()
+        if task == "get_activations":
+            response = self.get_activations(inputs)
+        elif task == "edit_activations":
+            response = self.edit_activations(inputs)
+        else:
+            response = self.generate(inputs)
```

Comment:
Just a suggestion for a cleaner way for this bit, ignore if you think this is unnecessary:)
```suggestion
        try:
            response = getattr(self, task)(inputs)
        except Exception as err:
            logger.error(f"Invalid task name: {err}")
```